lib/adapter.js

```diff
@@ -6940,18 +6924,23 @@ function Adapter(options) {
                             adapterObjects.subscribe(`${ALIAS_STARTS_WITH}*`);
                         }
 
-                        this._getObjectsByArray(aliasesIds, null, options, (errors, aliasObjs) =>
-                            aliasObjs.forEach(aliasObj =>
-                                this._addAliasSubscribe(aliasObj, aliasObj._id, () =>
-                                    !--count && typeof callback === 'function' && callback())));
+                        const aliasObjs = await new Promise(resolve => {
+                            this._getObjectsByArray(aliasesIds, null, options, (errors, aliasObjs) => {
+                                resolve(aliasObjs);
+                            });
+                        });
+
+                        aliasObjs.forEach(aliasObj => {
+                            promises.push(new Promise(resolve => this._addAliasSubscribe(aliasObj, aliasObj._id, resolve)));
+                        });
```

Comment:
Why not just await those in a loop? The containing function is now async.

Reply:
Because then they work unnecessarily sequential,  but yeah .. probably won't matter in this case

Reply:
I remember discussions that executing too many state updates at once can lead to issues - I'm guessing the same is true for objects. Maybe even earlier since object payloads are usually much larger.

Reply:
Good point, however `_addAliasSubscribe` is not updating objects or states in the db. As mentioned it probably won't hurt anyway, because normally this is not called thousands of times, so performance is not that relevant.

Reply:
we would need to replace the `forEach` loop and add a helper variable to check if there was an alias id for the final logic.. thus maybe it won't improve readability..

Reply:
Due to the above comments, I'll keep it as it is.